src/main/java/me/ryanhamshire/GPFlags/util/ChatUtil.java

```diff
@@ -0,0 +1,225 @@
+        message = message.replace("&m", "<strikethrough>");
+        message = message.replace("&n", "<underlined>");
+        message = message.replace("&o", "<italic>");
+        message = message.replace("&k", "<obfuscated>");
+        message = message.replace("&r", "<reset>");
+        message = message.replace("<red>", "<color:red>");
+        message = message.replace("<green>", "<color:green>");
+        message = message.replace("<blue>", "<color:blue>");
+        message = message.replace("<yellow>", "<color:yellow>");
+        message = message.replace("<aqua>", "<color:aqua>");
+        message = message.replace("<gold>", "<color:gold>");
+        message = message.replace("<gray>", "<color:gray>");
+        message = message.replace("<dark_gray>", "<color:dark_gray>");
+        message = message.replace("<dark_red>", "<color:dark_red>");
+        message = message.replace("<dark_green>", "<color:dark_green>");
+        message = message.replace("<dark_blue>", "<color:dark_blue>");
+        message = message.replace("<dark_aqua>", "<color:dark_aqua>");
+        message = message.replace("<dark_purple>", "<color:dark_purple>");
+        message = message.replace("<light_purple>", "<color:light_purple>");
+        message = message.replace("<black>", "<color:black>");
+        message = message.replace("<white>", "<color:white>");
+        message = message.replace("<b>", "<bold>");
+        message = message.replace("<u>", "<underlined>");
+        message = message.replace("<i>", "<italic>");
+        message = message.replace("<st>", "<strikethrough>");
+        message = message.replace("<obf>", "<obfuscated>");
+        return message;
+    }
+
+    // Validates a minimessage for a specific sender, meaning we check for permissions and remove components if necessary
+    private static String validateMinimessage(CommandSender sender, String message) {
+
+        /*
+        // Check if there is a link in the message without click-component
+        Pattern pattern = Pattern.compile("(?<!<click:open_url:)(https?://\\S+)(?!>)");
+        Matcher matcher = pattern.matcher(message);
+        while (matcher.find()) {
+            String link = message.substring(matcher.start(), matcher.end());
+            message = message.replace(link, "<click:open_url:" + link + ">" + link + "</click>");
+            matcher = pattern.matcher(message);
+        }*/
+
+        // Check if there are links
+        Pattern pattern = Pattern.compile("https?://(?:www\\.)?[\\w\\-.]+\\.[a-z]{2,}(?:/[\\w\\-.,@?^=%&:/~+#]*)?");
+        Matcher matcher = pattern.matcher(message);
+
+        String pseudoMessage = message;
+        while (matcher.find()) {
+            String link = pseudoMessage.substring(matcher.start(), matcher.end());
+            pseudoMessage = pseudoMessage.replace(link, "");
+            if(message.contains("<click:open_url:" + link + ">")) continue; // Skip if already a click-component
```

Comment:
What if there are two of the same links in the message? This'll only convert the first one

Reply:
Not anymore, same iteration as everywhere until there is no link available.